vault/seal/seal.go

```diff
@@ -513,10 +523,30 @@ const (
 
 // Encrypt uses the underlying seal to encrypt the plaintext and returns it.
 func (a *access) Encrypt(ctx context.Context, plaintext []byte, options ...wrapping.Option) (*MultiWrapValue, map[string]error) {
+	errs := make(map[string]error)
+
 	// Note that we do not encrypt with disabled wrappers. Disabled wrappers are only used to decrypt.
 	candidateWrappers := a.filterSealWrappers(enabledWrappers, healthyWrappers)
-	if len(candidateWrappers) == 0 {
-		// If all seals are unhealthy, try any way since a seal may have recovered
+	if len(candidateWrappers) > 0 {
+		// As there are healthy wrappers, add errors for any unhealthy ones, so that it
+		// it is clear that the resulting MultiWrapValue is missing ciphertext for some seals.
+		for i, unhealthyWrapper := range a.filterSealWrappers(enabledWrappers, unhealthyWrappers) {
```

Comment:
We could do this with a function that returns healthy and unhealthy in separate slices, but since this is single digit numbers this is clearer to read.